pyega3/libs/data_file.py

```diff
@@ -187,19 +187,24 @@ def download_file_slice(self, file_name, start_pos, length, options=None, pbar=N
 
         self.temporary_files.add(file_name)
 
-        existing_size = os.stat(file_name).st_size if os.path.exists(file_name) else 0
+        existing_size = os.stat(final_file_name).st_size if os.path.exists(final_file_name) else 0
         if existing_size > length:
-            os.remove(file_name)
+            os.remove(final_file_name)
+            existing_size = 0
+
         if pbar:
             pbar.update(existing_size)
 
         if existing_size == length:
-            return file_name
+            return final_file_name
 
         try:
-            with self.data_client.get_stream(path,
-                                             {
-                                                 'Range': f'bytes={start_pos + existing_size}-{start_pos + length - 1}'}) as r:
+            range_start = start_pos + existing_size
```

Comment:
Yes, it's much easier to understand like this! Great that you don't hesitate cleaning up the code, while you're at it!